x-pack/plugins/canvas/canvas_plugin_src/renderers/progress/progress.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { getId } from '../../../public/lib/get_id';
+import { shapes } from './shapes';
+
+export const progress = () => ({
+  name: 'progress',
+  displayName: 'Progress',
+  help: 'Reveal a percentage of an element',
+  reuseDomNode: true,
+  render(domNode, config, handlers) {
+    const { shape, value, max, valueColor, barColor, weight, label, labelPosition, font } = config;
+    const percent = value / max;
+    const shapeDef = shapes[shape];
+
+    if (shapeDef) {
+      const { path, getViewBox, width, height } = shapeDef;
```

Comment:
`shapeDef` is effectively a new API. Can you explain the parameters? Do we actually need getViewBox or can we simplify this?

Reply:
I added the `getViewBox` function to allow the user to define how the weight adjusts the shapes viewBox to keep full shape in view as the weight increase. I realized that I can apply a general calculation to all shapes to account for the stroke width, similar to how it's handled in the `shape` element.